src/bin/upgrade/main.rs

```diff
@@ -0,0 +1,107 @@
+//! `cargo upgrade`
+
+#![warn(missing_docs, missing_debug_implementations, missing_copy_implementations, trivial_casts,
+        trivial_numeric_casts, unsafe_code, unstable_features, unused_import_braces,
+        unused_qualifications)]
+
+extern crate docopt;
+extern crate pad;
+#[macro_use]
+extern crate serde_derive;
+extern crate toml;
+
+use std::error::Error;
+use std::io::{self, Write};
+use std::process;
+
+extern crate cargo_edit;
+use cargo_edit::{Manifest, get_latest_dependency};
+
+static USAGE: &'static str = r"
+Upgrade dependencies in a manifest file to the latest version.
+
+Usage:
+    cargo upgrade [--dependency <dep>...] [--manifest-path <path>]
+    cargo upgrade (-h | --help)
+    cargo upgrade (-V | --version)
+
+Options:
+    -h, --help                   Print this message
+    --dependency -d <dep>        Dependency to update
```

Comment:
Also it's worth mentioning here that upgrade will update dev, build and all target dependencies as well.